miriway.desktop.in

```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Name=Miriway
+Comment=Mir/Wayland desktop environment
+Exec=env MIRIWAY_SESSION_STARTUP=@CMAKE_INSTALL_FULL_LIBEXECDIR@/miriway-session-startup MIRIWAY_SESSION_SHUTDOWN=@CMAKE_INSTALL_FULL_LIBEXECDIR@/miriway-session-shutdown sh -l -c miriway
```

Comment:
I would recommend pulling this out into a wrapper script simply to ensure that we don't have problems with desktop file parsers in different login managers. It also allows you to set the TryExec to the same thing.

Perhaps call it `miriway-session`?

Reply:
Because unlike normal desktop files that are run from a desktop, this is run by a login manager and login managers don't necessarily have the most robust handlers for executing these lines.

Reply:
Even so:
```
$ grep Exec=env /usr/share/{wayland-sessions,xsessions}/*
/usr/share/wayland-sessions/miriway.desktop:Exec=env MIRIWAY_SESSION_STARTUP=/usr/libexec/miriway-session-startup MIRIWAY_SESSION_SHUTDOWN=/usr/libexec/miriway-session-shutdown sh -l -c miriway
/usr/share/wayland-sessions/ubuntu.desktop:Exec=env GNOME_SHELL_SESSION_MODE=ubuntu /usr/bin/gnome-session --session=ubuntu
/usr/share/wayland-sessions/ubuntu-wayland.desktop:Exec=env GNOME_SHELL_SESSION_MODE=ubuntu /usr/bin/gnome-session --session=ubuntu
/usr/share/xsessions/ubuntu.desktop:Exec=env GNOME_SHELL_SESSION_MODE=ubuntu /usr/bin/gnome-session --session=ubuntu
/usr/share/xsessions/ubuntu-xorg.desktop:Exec=env GNOME_SHELL_SESSION_MODE=ubuntu /usr/bin/gnome-session --session=ubuntu
```

Reply:
The GNOME_SHELL_SESSION_MODE is an Ubuntu thing: https://salsa.debian.org/gnome-team/gnome-session/-/blob/ubuntu/latest/debian/patches/ubuntu/ubuntu-sessions.patch?ref_type=heads

I don't have anything like this on my systems in Fedora.

Reply:
I guess? I thought snaps were fine with hyphenated binaries. It just looks weird to me to use a period there because it might get treated like a file extension.